### ISSUE TITLE
Expected class type but received "Overload[

### ISSUE THREAD

**Issue by AnatoliChe** [microsoft/pyright#3507]
Hello!
When I try lint code with type hint Value or Array from multiprocessing
like:
```
from multiprocessing import Value
def worker(complixity: Value):
        complixity.value += 1

complixity = Value('i', 0)
worker(complixity)
```
I have got lint error:

error: Expected class type but received "Overload[(typecode_or_type: Type[_CT@Value], *args: Any, lock: Literal[False]) -> _CT@Value, (typecode_or_type: Type[_CT@Value], *args: Any, lock: _LockLike | Literal[True]) -> SynchronizedBase[_CT@Value], (typecode_or_type: str, *args: Any, lock: _LockLike | Literal[True]) -> SynchronizedBase[Any], (typecode_or_type: str | Type[_CData], *args: Any, lock: _LockLike | bool = ...) -> Any]" (reportGeneralTypeIssues)

Maybe I do something wrong?
Maybe bad idea use static hints for shared memory? (@overloaded from multiprocessing)
Or I should say pyright do not use this check?

Could You help?
ps
Python 3.9.2
pyright 1.1.249

**Comment by erictraut** [comment]
Please include a minimal self-contained code sample. The sample above references a symbol called `Value`, and it's not declared or imported.

**Comment by erictraut** [comment]
`Value`, as its defined in the stub file `multiprocessing/__init__.pyi`, refers to an overloaded function, not a type. Functions cannot be used as a type annotation in the Python type system. That would be the equivalent of the following:

```python
def foo(): ...

def bar(x: foo): ...
```

If you think that `Value` is incorrectly defined, please file a bug in the [typeshed project](https://github.com/python/typeshed/issues). Pyright — and other type checkers in the Python world — consume the type stubs from typeshed.

Based on the type information provided to pyright, it's doing the right thing here.

**Comment by JelleZijlstra** [comment]
I was about to write something similar. For reference the relevant stubs are here: https://github.com/python/typeshed/blob/cb5b31cf15da14c63cb6467b5a29315471173ced/stdlib/multiprocessing/context.pyi#L76. In general, `multiprocessing` has a highly dynamic implementation that makes it difficult to use it safely with type checking.

**Comment by erictraut** [comment]
The answer depends on your goals. What is your motivation for including type annotations in your code? To improve completion suggestions when editing in a smart editor like VS Code? To provide human-readable documentation about types in your code? To eliminate bugs (and prevent future bugs) in your code through the use of static type checking? To detect bugs at runtime through the use of runtime type validation?

**Comment by erictraut** [comment]
If the only motivation is human readability, then a `# type: ignore` is a reasonable solution. But if you also want to find errors and provide better completion suggestions, then `# type: ignore` is not the right approach. You'd need to find or create a type that is legal to use in a type annotation. I don't know enough about the `multiprocessor` interface to offer more detailed advice. Perhaps the typeshed stub should define a callable protocol class and use that class as the annotated type of the `multiprocessing.context.BaseContext.Value` member.
